src/Evolution/Systems/RadiationTransport/M1Grey/M1Closure.cpp

```diff
@@ -0,0 +1,291 @@
+namespace M1Grey {
+
+void M1Closure(
+    const gsl::not_null<Scalar<DataVector>*> closure_factor,
+    const gsl::not_null<tnsr::II<DataVector, 3, Frame::Inertial>*>
+        pressure_tensor,
+    const gsl::not_null<Scalar<DataVector>*> comoving_energy_density,
+    const gsl::not_null<Scalar<DataVector>*> comoving_momentum_density_normal,
+    const gsl::not_null<tnsr::i<DataVector, 3, Frame::Inertial>*>
+        comoving_momentum_density_spatial,
+    const Scalar<DataVector>& energy_density,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& momentum_density,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& fluid_velocity,
+    const Scalar<DataVector>& fluid_lorentz_factor,
+    const tnsr::ii<DataVector, 3, Frame::Inertial>& spatial_metric,
+    const tnsr::II<DataVector, 3, Frame::Inertial>&
+        inv_spatial_metric) noexcept {
+  // Small number used to avoid divisions by zero
+  static constexpr double avoid_divisions_by_zero = 1.e-150;
+  // Below small_velocity, we use the v=0 closure,
+  // and we do not differentiate between fluid/inertial frames
+  static constexpr double small_velocity = 1.e-15;
+  // Dimension of spatial tensors
+  constexpr size_t spatial_dim = 3;
+  // Number of significant digits used in the rootfinding rooting
+  // used to find the closure factor
+  constexpr size_t root_find_number_of_digits = 6;
+  Variables<tmpl::list<
+      hydro::Tags::LorentzFactorSquared<DataVector>, MomentumSquared,
+      MomentumUp,
+      hydro::Tags::SpatialVelocityOneForm<DataVector, 3, Frame::Inertial>,
+      hydro::Tags::SpatialVelocitySquared<DataVector>>>
+      temp_closure_tensors(get(energy_density).size());
+
+  // The main calculation needed for the M1 closure is to find the
+  // roots of J^2 zeta^2 = H^a H_a, with J the fluid-frame energy density
+  // and H^a the fluid-frame momentum density (0th and 1st moments).
+
+  // We first compute various fluid quantities and contractions
+  // with inertial moments needed even if v^2 is small
+  auto& w_sqr =
+      get<hydro::Tags::LorentzFactorSquared<DataVector>>(temp_closure_tensors);
+  get(w_sqr) = square(get(fluid_lorentz_factor));
+  auto& v_sqr = get<hydro::Tags::SpatialVelocitySquared<DataVector>>(
+      temp_closure_tensors);
+  get(v_sqr) = 1. - 1. / get(w_sqr);
+  // S^i, the neutrino momentum tensor
+  auto& s_M = get<MomentumUp>(temp_closure_tensors);
+  raise_or_lower_index(make_not_null(&s_M), momentum_density,
+                       inv_spatial_metric);
+  // S^i S_i
+  auto& s_sqr = get<MomentumSquared>(temp_closure_tensors);
+  dot_product(make_not_null(&s_sqr), s_M, momentum_density);
+  // v_i, the spatial velocity one-form of the fluid
+  auto& v_m =
+      get<hydro::Tags::SpatialVelocityOneForm<DataVector, 3, Frame::Inertial>>(
+          temp_closure_tensors);
+  raise_or_lower_index(make_not_null(&v_m), fluid_velocity, spatial_metric);
+
+  // Allocate memory for H^i
+  tnsr::I<double, 3, Frame::Inertial> H_M(0.);
```

Comment:
In the past I remember we tried to make variable names more explicit (as an example, here I would have expected us to request `w` -> `lorentz_factor`,  `_M` -> `_upper`, `_m` -> `_lower`, and so on). In particular, `H_M` does not seem compatible with our style guide. Are we still trying to follow those guidelines/suggestions, or is my understanding out of date?